pootle/apps/pootle_app/management/commands/project_serializers.py

```diff
@@ -0,0 +1,189 @@
+            help="Clear serializers for project")
+        parser.add_argument(
+            '--clear-deserializers',
+            action='store_true',
+            dest='clear_deserializers',
+            help="Clear deserializers for project")
+
+    def handle_all(self, **options):
+        if options["list"]:
+            return self.print_serializers_list()
+        has_setters = (
+            options["serializers"]
+            or options["deserializers"]
+            or options["clear_serializers"]
+            or options["clear_deserializers"])
+        if has_setters and not options["project"]:
+            raise CommandError(
+                _("You must specify a project to set the serializers "
+                  "or deserializers for"))
+        if options["project"]:
+            return self.handle_project(**options)
+        for options["project"] in Project.objects.exclude(disabled=True):
+            self.print_project_serialization(options["project"])
+
+    def print_serializers_list(self):
+        serials = serializers.gather(Store)
+        if not serials.keys():
+            self.stdout.write(
+                "There are no serializers set up on your system")
+
+        deserials = deserializers.gather(Store)
+        if not deserials.keys():
+            self.stdout.write(
+                "There are no deserializers set up on your system")
+
+        if serials.keys():
+            heading = "Serializers"
+            self.stdout.write("\n%s" % heading)
+            self.stdout.write("-" * len(heading))
+            for name, serializer in serials.items():
+                self.stdout.write(
+                    "{: <30} {: <25} ".format(name, serializer))
+        if deserials.keys():
+            heading = "Deserializers"
+            self.stdout.write("\n%s" % heading)
+            self.stdout.write("-" * len(heading))
+            for name, deserializer in deserials.items():
+                self.stdout.write(
+                    "{: <30} {: <25} ".format(name, deserializer))
+
+    def print_project_serialization(self, project):
+        project_name = "Project: %s(%s)" % (project.name, project.code)
+        self.stdout.write(project_name)
+        self.stdout.write(
+            "  local filetype: %s" % project.localfiletype)
+        self.stdout.write(
+            "  serializers: %s" % (project.serializers or ""))
+        self.stdout.write(
+            "  deserializers: %s" % (project.deserializers or ""))
+
+    def validate_deserializers(self, deserializerstr):
+        avail_deserials = deserializers.gather(Store).keys()
+        deserials = [
+            x.strip() for x in deserializerstr.split(",") if x.strip()]
+        for deserial in deserials:
+            if deserial not in avail_deserials:
+                raise CommandError(
+                    "Missing deserializer(%s): "
+                    "This deserializer is not installed on your system" % deserial)
+        return ",".join(deserials)
```

Comment:
Likely better pushed up to be an `options=` setting in Argparse.  That lets argparse catch this.